.golangci.yml

```diff
@@ -1,253 +1,178 @@
-      # those files from linting, so this linter is not useful.
-      - fieldalignment
-      # Disable shadow
-      - shadow
-  perfsprint:
-    # Optimizes even if it requires an int or uint type cast.
-    int-conversion: true
-    # Optimizes into `err.Error()` even if it is only equivalent for non-nil errors.
-    err-error: true
-    # Optimizes `fmt.Errorf`.
-    errorf: true
-    # Optimizes `fmt.Sprintf` with only one argument.
-    sprintf1: true
-    # Optimizes into strings concatenation.
-    strconcat: true
-  revive:
-      ignore-generated-header: true
+  disable:
+    - errcheck
+  settings:
+    depguard:
+      rules:
+        disallow-crossdock:
+          files:
+            - '!**/crossdock/**'
+          deny:
+            - pkg: github.com/crossdock/crossdock-go
+              desc: Do not refer to crossdock from other packages
+        disallow-otel-contrib-translator:
+          files:
+            - '!**/v1adapter/**'
+          deny:
+            - pkg: github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger
+              desc: Use v1adapter package instead of opentelemetry-collector-contrib/pkg/translator/jaeger
+        disallow-uber/goleak:
+          files:
+            - '**_test.go'
+          deny:
+            - pkg: go.uber.org/goleak
+              desc: Use github.com/jaegertracing/jaeger/internal/testutils
+        disallowed-deps:
+          deny:
+            - pkg: go.uber.org/atomic
+              desc: Use 'sync/atomic' instead of go.uber.org/atomic
+            - pkg: io/ioutil
+              desc: Use os or io instead of io/ioutil
+            - pkg: github.com/hashicorp/go-multierror
+              desc: Use errors.Join instead of github.com/hashicorp/go-multierror
+            - pkg: go.uber.org/multierr
+              desc: Use errors.Join instead of github.com/hashicorp/go-multierror
+            - pkg: github.com/jaegertracing/jaeger/model$
+              desc: Use github.com/jaegertracing/jaeger-idl/model/v1
+            - pkg: github.com/jaegertracing/jaeger/proto-gen/api_v2$
+              desc: Use github.com/jaegertracing/jaeger-idl/proto-gen/api_v2
+    gosec:
+      excludes:
+        - G104
+        - G107
+        - G404
+        - G601
+    govet:
+      disable:
+        - fieldalignment
+        - shadow
+      enable-all: true
+    perfsprint:
+      int-conversion: true
+      err-error: true
+      errorf: true
+      sprintf1: true
+      strconcat: true
+    revive:
       severity: error
       enable-all-rules: true
-      # See https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md
       rules:
-        # not a completely bad linter, but needs clean-up and sensible width (80 is too small)
         - name: line-length-limit
+          arguments:
+            - 80
           disabled: true
-          arguments: [80]
-        # would be ok if we could exclude the test files, but otherwise too noisy
```

Comment:
please restore all comments for disabled linters because they explain why we disabled them

Reply:
you can just copy these settings from the original file and maybe change indentation to match new config